### DISCUSSION TITLE
Can `tokio` use libaco to get a better performance ?

### DISCUSSION THREAD

**Discussion by loongs-zhang** [tokio-rs/tokio#4482] [Ideas] [answered]
[libaco](https://github.com/hnes/libaco) is a `c coroutine library`, the question is shown in the title.

**Comment by Noah-Kennedy** [comment, marked as answer]
Tokio uses the futures/coroutine model built into rustc and the standard library. So we can't exactly switch to a different coroutine model. Also, I would be careful when analyzing benchmarks on coroutine switching, as the relevance of such benchmarks to real-world application performance is often limited.
